parquet-hadoop/src/test/java/org/apache/parquet/hadoop/codec/TestLz4RawCodec.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.hadoop.codec;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.compress.CompressionOutputStream;
+import org.apache.parquet.bytes.BytesInput;
+import org.apache.parquet.hadoop.codec.*;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Random;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+public class TestLz4RawCodec {
```

Comment:
I'm curious, why are you creating dedicated test cases for LZ4 instead of sharing the same basic tests for all compressors/decompressors?

Reply:
That would be ideal. However, the test folder already has dedicated test cases for different codec, e.g. TestSnappyCodec and TestZstandardCodec. They do not use common or abstract class in the implementation so it is non-trivial to make a generic test for all codecs at the moment.

Reply:
Well, writing a dedicated test is non-trivial either :-) And the more you pile up dedicated tests, the less trivial it will be to reconcile them...

Reply:
You could at least write this one as a generic test, specialize it for all existing codecs, while also keeping the existing dedicated tests to ensure we don't lose coverage. Does that sound reasonable?

Reply:
> You could at least write this one as a generic test, specialize it for all existing codecs, while also keeping the existing dedicated tests to ensure we don't lose coverage. Does that sound reasonable?

I agree. I have changed the test to be generic to use only CompressionCodec, Compressor and Decompressor in my latest patch. For now I only enable LZ4 RAW in the test due to missing JNI library on my Mac for verification. Should be easy to extend for other codecs. Please take a look, thanks! @pitrou